servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/ProxyConnectConnectionFactoryFilter.java

```diff
@@ -87,7 +93,13 @@ public Single<C> newConnection(final ResolvedAddress resolvedAddress,
                 return delegate().newConnection(resolvedAddress, contextMap, observer).flatMap(c -> {
                     try {
                         return c.request(c.connect(connectAddress).addHeader(CONTENT_LENGTH, ZERO))
-                                .flatMap(response -> handleConnectResponse(c, response).shareContextOnSubscribe())
+                                // Successful response to CONNECT never has a message body, and we are not interested in
+                                // payload body for any non-200 status code. Drain it asap to free connection and RS
+                                // resources before starting TLS handshake.
+                                .flatMap(response -> response.messageBody().ignoreElements()
+                                        .concat(Single.defer(() -> handleConnectResponse(c, response)
```

Comment:
Tried to minimize git diff, added draining here instead of changing indentation inside `handleConnectResponse`